src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -379,6 +379,14 @@
     ),
     # This is a placeholder for other dialects.
     SimpleArrayTypeGrammar=Nothing(),
+    BaseExpressionElementGrammar=OneOf(
```

Comment:
Extracted this from `SelectClauseElementSegment` because the BigQuery dialect wants to extend it.

Reply:
Sounds very reasonable 👍